forui/lib/src/widgets/pagination/pagination_controller.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:flutter/widgets.dart';
+
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+
+/// A controller that controls which page is selected.
+class FPaginationController extends FChangeNotifier {
+  /// The number of sibling pages displayed beside the current page number. Defaults to 1.
+  ///
+  /// This value determines how many pages are shown on either side of the
+  /// currently selected page in the pagination.
+  ///
+  /// For example, if `siblings` is 2 and the current page is 5,
+  /// the displayed pages would be 3, 4, 5, 6, and 7.
+  ///
+  /// # Contract:
+  /// * Throws [AssertionError] if [siblings] < 0.
+  final int siblings;
+
+  /// Whether to show the first and last pages in the pagination. Defaults to `true`.
+  ///
+  /// If `true`, the pagination will always display the first and last page, regardless of the current page.
+  /// This can be useful for allowing users to quickly navigate to the beginning or end of the paginated content.
+  final bool showEdges;
+
+  /// A callback that is triggered whenever the current page changes.
+  ///
+  /// This callback is invoked whenever the `page` property of the `FPaginationController`
+  /// is updated, either programmatically or through user interaction.
+  ///
+  /// This can be useful for performing actions in response to page changes,
+  /// such as updating other parts of the UI or triggering analytics events.
+  final VoidCallback? onPageChanged;
```

Comment:
The documentation seems to cut-off at varying line widths. Please be consistent and cut it off at 120 characters.